cpp/src/copying/slice.cpp

```diff
@@ -31,30 +33,29 @@ std::vector<column_view> slice(column_view const& input,
 {
   CUDF_EXPECTS(indices.size() % 2 == 0, "indices size must be even");
 
-  if (indices.empty()) { return {}; }
+  if (indices.empty()) return {};
 
   auto null_counts = cudf::detail::segmented_count_unset_bits(input.null_mask(), indices, stream);
 
-  std::vector<column_view> children{};
-  for (size_type i = 0; i < input.num_children(); i++) { children.push_back(input.child(i)); }
+  auto f = cudf::test::make_counting_transform_iterator(0, [&](auto i) { return input.child(i); });
```

Comment:
Again seems like child_begin/child_end removes need for a transform iterator entirely here?